base/sequence_clock.go

```diff
@@ -291,6 +292,23 @@ func (c *SequenceClockImpl) UpdateWithClock(updateClock SequenceClock) {
 	}
 }
 
+func (c *SequenceClockImpl) UpdateWithPartitionClocks(partitionClocks []*PartitionClock, allowRollback bool) {
+
+	for _, partitionClock := range partitionClocks {
+		if partitionClock != nil {
+			for vbNo, seq := range *partitionClock {
+				if seq > 0 {
+					if seq < c.value[vbNo] && !allowRollback {
+						panic(fmt.Sprintf("Update attempted to set clock to earlier sequence.  Vb: %d, currentSequence: %d, newSequence: %d", vbNo, c.value[vbNo], seq))
```

Comment:
rather than panicking, I think this should return an error and have it propagate up to a higher level function. This will make it much easier to unit test among other things.

For now since we're running behind on deadline, can you just add a TODO comment and I'll try to get a PR up next time I'm in this code?